### PR TITLE
BUG: _read_inline_image infinite loop

### PR DESCRIPTION
**Pull request by yuvalgtrustmi**
fixes https://github.com/py-pdf/pypdf/issues/2423

### CODE REVIEW

**Comment by stefan6419846** [comment]
Thanks for the PR. Are you able to provide a corresponding test which ensures that this will not break again in the future?

File: pypdf/generic/_data_structures.py [hunk 1]
```diff
@@ -1068,10 +1068,12 @@ def _read_inline_image(self, stream: StreamType) -> Dict[str, Any]:
             # We have reached the end of the stream, but haven't found the EI operator.
             if not buf:
                 raise PdfReadError("Unexpected end of stream")
+            if len(buf) == 2 and buf == b'EI':
```
Comment (pubpub-zz) [inline pypdf/generic/_data_structures.py:1071]: is the `len(buf) == 2` required ? `buf == b'EI'` is not sufficient ?

Comment (yuvalgtrustmi) [reply pypdf/generic/_data_structures.py:1071]: Absolutely

File: pypdf/generic/_data_structures.py [hunk 2]
```diff
@@ -1068,6 +1068,8 @@ def _read_inline_image(self, stream: StreamType) -> Dict[str, Any]:
             # We have reached the end of the stream, but haven't found the EI operator.
             if not buf:
                 raise PdfReadError("Unexpected end of stream")
+            if buf == b'EI':
```
Comment (MartinThoma) [inline pypdf/generic/_data_structures.py:1071]:
```suggestion
            if buf == b"EI":
```

**Review by pubpub-zz — COMMENTED** [review]
a test is more than required
it could be built splitting the content of a pdf 

- [ ] into 3 parts, the second containing "EI" only

**Comment by stefan6419846** [comment]
@yuvalgtrustmi Are you still interested in working on this PR and adding corresponding tests?

**Comment by pubpub-zz** [comment]
@yuvalgtrustmi 
I've produced a Pdf where the content has been cut 3 parts with one "subcontent" with only EI:
[EI_isolated_in_content.pdf](https://github.com/py-pdf/pypdf/files/14397674/tt.pdf)
however I can find how to reproduce your issue. Can you provide the code you've used to initially produce the error

**Comment by pubpub-zz** [comment]
> @yuvalgtrustmi any update?

any update ? On my side I can not reproduce the issue

**Comment by pubpub-zz** [comment]
@yuvalgtrustmi 
last request before closing the PR for cleanup. It can be reopened later

**Comment by pubpub-zz** [comment]
I close this PR and associated issue for clean-up.feel free to ask to reopen it if you have more data

**Pull request closed without merging**